pennylane/ops/op_math/prod.py

```diff
@@ -466,14 +466,19 @@ def terms(self):
         """
         # try using pauli_rep:
         if pr := self.pauli_rep:
-            ops = [pauli.operation() for pauli in pr.keys()]
+            with qml.QueuingManager.stop_recording():
+                ops = [pauli.operation() for pauli in pr.keys()]
             return list(pr.values()), ops
 
-        global_phase, factors = self._simplify_factors(factors=self.operands)
+        with qml.QueuingManager.stop_recording():
+            global_phase, factors = self._simplify_factors(factors=self.operands)
 
         factors = list(itertools.product(*factors))
 
-        factors = [Prod(*factor).simplify() if len(factor) > 1 else factor[0] for factor in factors]
+        with qml.QueuingManager.stop_recording():
```

Comment:
not sure how costly context managers are, but you can just evaluate `factors` within the first `stop_recording` context to not switch in and out twice